docs/templates/parameters.md

```diff
@@ -196,6 +196,53 @@ variable "cpu" {
 
 > ⚠️ Legacy (`variable`) parameters and rich parameters can't be used in the same template.
 
+### Migration
+
+Terraform variables shouldn't be used for workspace scoped parameters anymore, and it's recommended to convert variables to `coder_parameter` resources. To make the migration smoother, there is a special property introduced -
+`legacy_variable` and `legacy_variable_name` , which can link `coder_parameter` with a legacy variable.
+
+```hcl
+variable "legacy_cpu" {
+  sensitive   = false
+  description = "CPU cores"
+  default     = 2
+}
+
+data "coder_parameter" "cpu" {
+  name        = "CPU cores"
+  type        = "number"
+  description = "Number of CPU cores"
+  mutable     = true
+
+  legacy_variable_name = "legacy_cpu"
+  legacy_variable = var.legacy_cpu
+}
+```
+
+#### Steps
+
+1. Prepare and update a new template version:
+
+- Add `coder_parameter` resource matching the legacy parameter to migrate.
```

Comment:
```suggestion
- Add `coder_parameter` resource matching the legacy variable to migrate.
```

Should we expand on what "matching" means in this context? This might be confusing for the user.